evap/student/forms.py

```diff
@@ -12,8 +12,45 @@ def __init__(self, label):
         super().__init__(label=label, required=False)
 
 
+class TextAnswerField(forms.CharField):
+
+    def __init__(self, related_question_id="", *args, **kwargs):
+        # pylint: disable=keyword-arg-before-vararg
+        self.related_question_id = related_question_id
+        kwargs['required'] = False
+        kwargs['widget'] = forms.Textarea(attrs={'related_question_id': self.related_question_id})
+        super().__init__(*args, **kwargs)
+
+    def deconstruct(self):
+        name, path, args, kwargs = super().deconstruct()
+        del kwargs['required']
+        if self.related_question_id != "":
+            kwargs['related_question_id'] = self.related_question_id
+        return name, path, args, kwargs
+
+
+class RatingAnswerField(forms.TypedChoiceField):
+
+    def __init__(self, widget_choices, has_additional_textanswer=False, *args, **kwargs):
```

Comment:
i would suggest `allows_textanswer`, `has_additional_textanswer_field`, or `allows_additional_textanswer` as parameter name.

Reply:
went for `allows_textanswer`